src/Nokia_LCD.cpp

```diff
@@ -263,11 +263,27 @@ bool Nokia_LCD::print(unsigned long number) {
     return print(number_as_string);
 }
 
+//----
+// 27-Nov-2020 - Correction of issue #6 - Error in print(float) function
+//
+// In previous release, numbers like 1.0123 or 1.00123 were printed 1.123,
+// having the 0s next to the coma skipped.
+//----
 bool Nokia_LCD::print(double number, unsigned short decimals) {
     double integral = 0;
     long fractional = pow(10.0, decimals) * modf(number, &integral);
+    bool out_of_bounds = false;
 
-    return print(static_cast<long>(number)) || print(".") || print(fractional);
```

Comment:
The introduced code isn't equivalent to the previous one in regards to what happens when we get an out of bounds error.

Specifically, before, if one of those `print` statements would return `true`, then the entire expression would be lazily evaluated to `true` and the remaining statements would never execute. In the suggested case, the out of bounds flag will be correctly raised, however, the out of bounds text will keep being printed.
I don't remember what actually happens in that case and whether the cursor goes back to the beginning but this can potentially be a problem.

Reply:
Nice fix! Took me a second to realize how `out_of_bounds` is now used but it makes sense! :grin: